homeassistant/components/planifneige/__init__.py

```diff
@@ -0,0 +1,114 @@
+"""
+Platform for the City of Montreal's Planif-Neige snow removal APIs.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.planifneige/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.helpers.discovery import async_load_platform
+from homeassistant.helpers.dispatcher import dispatcher_send
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import (CONF_NAME, CONF_API_KEY,
+                                 CONF_UPDATE_INTERVAL, CONF_SCAN_INTERVAL,
+                                 CONF_UPDATE_INTERVAL_INVALIDATION_VERSION)
+
+DOMAIN = 'planifneige'
+
+DATA_UPDATED = '{}_data_updated'.format(DOMAIN)
+
+REQUIREMENTS = ['planif-neige-client==0.0.5']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_STREETID = 'streetid'
+CONF_STREETS = 'streets'
+CONF_DBPATH = 'database_path'
+
+ATTR_SENSOR = 'sensor'
+ATTR_STATES = 'states'
+
+DEFAULT_INTERVAL = timedelta(minutes=5)
+
+_STREET_SCHEME = vol.Schema({
+    vol.Required(CONF_NAME): cv.string,
+    vol.Required(CONF_STREETID): cv.positive_int
+})
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.All(
+        vol.Schema({
+            vol.Required(CONF_API_KEY): cv.string,
+            vol.Optional(CONF_SCAN_INTERVAL, default=DEFAULT_INTERVAL):
+                vol.All(cv.time_period, cv.positive_timedelta),
+            vol.Required(CONF_DBPATH): cv.string,
```

Comment:
Requiring users to define a path to a file (which doesn't exist yet) seem like exposing too much internal details. Maybe exposing an optional `CONF_FILE_PATH` config entry to where a database would then be created with a consistent name by the platform would hide such inner workings while leaving the flexibility to the user of optionally defining a sub-folder.

Reply:
The path to the database should default to HA's configuration directory and the DB created there.